### PR TITLE
Tosh-Mackrine/#85/view user profile data

### LINKED ISSUE #85: view user profile data
Add a query that allows user to view their profile data.
The query should return:

- [ ] Email
- [ ] Name
- [ ] Phone
- [ ] Reason for joining
- [ ] Date Joined

Ensure you add relevant tests for the feature. (Both Integration and Unit)
Ensure you also add relevant documentation of the query that you create.

### PR DESCRIPTION
**Pull request by Muriithi-Gitonga**
#### This PR contains:
-  User profile query and the tests
- Documentation for the query

### CODE REVIEW

File: api/spec/graphql/query/profile_spec.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    describe 'get profile' do
+        context 'when user exists' do
+            before do
+                post "/graphql", params: { query: query(id: user.id) }
+            end
+
+            it 'returns a success message' do
+                json = JSON.parse(response.body)
+                pp json
```
Comment (otsembo) [inline api/spec/graphql/query/profile_spec.rb]: Remove unnecessary print.

File: api/spec/graphql/query/profile_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                expect(json['data']['profile']['message']).to eq('user profile fetched successfully')
+            end
+
+            it 'returns a success status' do
+                json = JSON.parse(response.body)
+                pp json
```
Comment (otsembo) [inline api/spec/graphql/query/profile_spec.rb]: Remove unnecessary print.

File: api/db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+    name: "Tosh",
+    phone: "123456789",
+    password: "Password1.",
+    date_of_birth: "1990-01-01",
+    reason_for_joining: 1,
+    isMuslim: true,
+
+)
+
+puts "Done seeding"
```
Comment (otsembo) [inline api/db/seeds.rb]: We do not need any seeding data in production for now. You can omit this from version control for now.

File: api/spec/graphql/query/profile_spec.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
+RSpec.describe 'user mutation', type: :request do
```
Comment (otsembo) [inline api/spec/graphql/query/profile_spec.rb]: You are describing a `query` as opposed to `mutation`. Kindly fix the naming.

**Review by otsembo — CHANGES_REQUESTED** [review]
Hello @Muriithi-Gitonga @mackrineawino Great job on the PR.
I have left a few comments on things you can fix.

@Kimutai01 @naomirono Kindly make your reviews on this.

**Comment by mackrineawino** [comment]
> Hello @Muriithi-Gitonga @mackrineawino Great job on the PR. I have left a few comments on things you can fix.
> 
> @Kimutai01 @naomirono Kindly make your reviews on this.

Hello @otsembo  we have made the requested changes. Kindly check.

**Review by Kimutai01 — APPROVED** [review]
hello @Muriithi-Gitonga and @mackrineawino,
Good job on making the changes requested before. 
Everything works well

**Review by otsembo — APPROVED** [review]
Everything looks good now. Great job @Muriithi-Gitonga @mackrineawino

**Review by naomirono — APPROVED** [review]

**Pull request merged**